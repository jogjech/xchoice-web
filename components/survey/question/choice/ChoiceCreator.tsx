import React, { FunctionComponent } from "react";
import { Form, Input, Button } from "antd";
import { FormListFieldData } from "antd/lib/form/FormList";
import { PlusOutlined } from "@ant-design/icons";

interface Props {
  questionField: FormListFieldData;
}

const ChoiceCreator: FunctionComponent<Props> = ({ questionField }) => {
  return (
    <Form.List name={[questionField.name, "choices"]}>
      {(fields, { add: addChoice, remove: removeChoice }) => (
        <>
          {fields.map((field) => (
            <div key={field.key}>
              <Form.Item
                {...field}
                name={[field.name, "text"]}
                fieldKey={[field.fieldKey, "text"]}
                label={`Choice ${field.key + 1}`}
                rules={[{ required: true, message: "Missing choice" }]}
              >
                <Input placeholder="Choice" maxLength={100} />
              </Form.Item>
            </div>
          ))}
          <Form.Item>
            <Button
              type="dashed"
              onClick={() => addChoice()}
              block
              icon={<PlusOutlined />}
            >
              Add choice
            </Button>
          </Form.Item>
        </>
      )}
    </Form.List>
  );
};

export default ChoiceCreator;
