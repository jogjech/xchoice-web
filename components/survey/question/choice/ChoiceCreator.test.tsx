import React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import { FormListFieldData } from "antd/lib/form/FormList";
import ChoiceCreator from "./ChoiceCreator";

describe("Choice creator", () => {
  let wrapper: ShallowWrapper;
  const questionField: FormListFieldData = {
    name: 2,
    key: 3,
    fieldKey: 7,
  };

  beforeEach(() => {
    wrapper = shallow(
      <ChoiceCreator questionField={questionField}></ChoiceCreator>
    );
  });

  it("renders correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("contains form list", () => {
    const formList = wrapper.find("FormList");
    expect(formList.prop("name")).toEqual([questionField.name, "choices"]);
  });
});
