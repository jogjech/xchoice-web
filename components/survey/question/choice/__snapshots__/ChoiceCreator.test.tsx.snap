// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Choice creator renders correctly 1`] = `
<FormList
  name={
    Array [
      2,
      "choices",
    ]
  }
>
  <Component />
</FormList>
`;
